eppo/src/main/java/cloud/eppo/android/ConfigurationRequestor.java

```diff
@@ -24,45 +24,50 @@ public ConfigurationRequestor(ConfigurationStore configurationStore, EppoHttpCli
     }
 
     public void load(InitializationCallback callback) {
-        AtomicBoolean cachedUsed = new AtomicBoolean(false);
+        AtomicBoolean callbackCalled = new AtomicBoolean(false);
         configurationStore.loadFromCache(new CacheLoadCallback() {
             @Override
             public void onCacheLoadSuccess() {
-                cachedUsed.set(true);
-                if (callback != null) {
+                if (callback != null && callbackCalled.compareAndSet(false, true)) {
```

Comment:
[`compareAndSet()`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicBoolean.html#compareAndSet-boolean-boolean-) is atomic and should prevent any race conditions resulting in multiple threads each thinking the other hasn't yet called the callback